### ISSUE TITLE
InputActionHandler's OnInputActionEnded fires twice despite OnInputActionStart firing once

### ISSUE THREAD

**Issue by tphan25** [microsoft/MixedRealityToolkit-Unity#7733]
## Describe the bug

When using an InputActionHandler for "Select", without Focus Required and attached to an empty Game Object, I click Play in the Unity Editor and use my Mixed Reality Headset (Lenovo Explorer) + a single controller, with DefaultMixedRealityToolkitConfigurationProfile. When I click the trigger button quickly, (corresponding to a Select action on the Immersive Headset controllers) it fires off the OnInputActionStarted event once, and fires the OnInputActionEnded event twice. **Oddly enough, if I hold for a few seconds and release then it performs the proper behavior and only fires off once.** 

## To reproduce

1. Using an Immersive/VR Headset for Mixed Reality, create a scene and add MixedRealityToolkit to it with DefaultMixedRealityToolkitConfigurationProfile.
2. Add an empty GameObject, and attach an InputActionHandler and a MonoBehavior script to it. 
3. In the MonoBehavior, add two methods, one for attaching to OnInputActionStarted and one for OnInputActionEnded.
4. In the InputActionHandler, uncheck IsFocusRequired and make sure InputAction is "Select".
4. Drag the empty GameObject into the InputActionHandler's object fields and make it call your MonoBehavior's scripts for start and end respectively.
3. Click Play in the editor. When I click the trigger button on an Immersive/VR Headset, the event for action end fires twice and for action start fires once.

## Expected behavior

For a single InputActionHandler, OnInputActionStarted should fire once and OnInputActionEnded should fire once.

## Screenshots

**Inspector Window for GameObject**
![image](https://user-images.githubusercontent.com/25301834/80335326-f5f51780-8821-11ea-9ea4-7d4af58b2807.png)
**MonoBehavior Script**
![image](https://user-images.githubusercontent.com/25301834/80335366-158c4000-8822-11ea-9058-d2004f5821b3.png)
**Console Logging from example script**
![image](https://user-images.githubusercontent.com/25301834/80335527-8af81080-8822-11ea-9eb1-6de1fba73334.png)

## Your setup (please complete the following information)

- Unity Version 2019.3.7f1
- MRTK Version 2.3.0
- WMR Immersive with Headset but using one controller since my other one did not have battery at the time 💀 

## Target platform (please complete the following information)

- WMR immersive

## Additional context
Was basically trying to implement a virtual keyboard and wanted to only type keys once I released the trigger. However, I discovered this when I moved some logic to OnInputActionEnd it would type a character twice (because of this issue). I thought it might have had something to do with the parent object also having an InputActionHandler but since I can reproduce here and it also had the same issue when disabling the parent's handler it seems like that was not the case.

**Comment by davidkline-ms** [comment]
@MaxWang-MS, can you investigate this issue and look for a fix?

**Comment by MaxWang-MS** [comment]
Hi @tphan25 could you confirm whether the issue still occurs with the latest version of MRTK and a relatively new version of Unity 2019.4?

**Comment by keveleigh** [comment]
More information at https://github.com/microsoft/MixedRealityToolkit-Unity/issues/8776 (these sound like the same issue).

The `OnInputAction*` events gather all other fired input events and send them along. Currently, with the default legacy profiles, we support the "select"/"tap" gesture, which raises a gesture completed event at the time of recognition. This would be why you aren't seeing the double event when holding longer, since "select" has a time component and doesn't fire if that timespan has elapsed.

Basically, you're seeing `OnGestureCompleted` and `OnInputUp`, both with "Select" actions, coming across the `OnInputActionEnded` event. I'd recommend removing select from the gesture profile if this is an issue.
